TheApiAwakens/StarWarsViewController.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  ViewController.swift
+//  TheApiAwakens
+//
+//  Created by Tassia Serrao on 12/01/2017.
+//  Copyright © 2017 Tassia Serrao. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController, UIGestureRecognizerDelegate {
+
+    @IBOutlet weak var vehicleStackView: UIStackView!
+    @IBOutlet weak var characterStackview: UIStackView!
+    @IBOutlet weak var starshipStackview: UIStackView!
+    var type: ResourceType!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let stackViews = [characterStackview, vehicleStackView, starshipStackview]
+        
+        for stackView in stackViews {
+            let recognizer = UITapGestureRecognizer(target: self, action: #selector(showSWDetailViewController(_:)))
+            stackView?.addGestureRecognizer(recognizer)
```

Comment:
That's a really good question. It made me think about it. I believe StackView is optional because they are IBOutlets. When you declare an Outlet the compiler automatically converts the type to an implicitly unwrapped optional and sets the value to nil. So, even though the compiler guarantees that StackView will not be nil once I need it, we are dealing with Views that were nil and could be nil. I can even replace ! for ? on my ViewControler when  I declare an IBOutlet.